src/CodeGenerator/Domain/FilenameSanitizer.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gacela\CodeGenerator\Domain;
+
+use RuntimeException;
+
+final class FilenameSanitizer
+{
+    public const FACADE = 'Facade';
+    public const FACTORY = 'Factory';
+    public const CONFIG = 'Config';
+    public const DEPENDENCY_PROVIDER = 'DependencyProvider';
+
+    public const EXPECTED_FILENAMES = [
+        self::FACADE,
+        self::FACTORY,
+        self::CONFIG,
+        self::DEPENDENCY_PROVIDER,
+    ];
+
+    public static function expectedFilenames(string $glue = ', '): string
+    {
+        return implode($glue, self::EXPECTED_FILENAMES);
+    }
+
+    public function sanitize(string $filename): string
+    {
+        $percents = [];
+        foreach (self::EXPECTED_FILENAMES as $expected) {
+            $percents[$expected] = similar_text($expected, $filename, $percent);
+        }
+
+        $maxVal = max($percents);
+        $maxValKeys = array_keys($percents, $maxVal, true);
```

Comment:
This is the magic! https://www.php.net/manual/en/function.similar-text.php
> similar_text — Calculate the similarity between two strings

Here I compare the similarity between the incoming filename and the potentially expected, and then the one with more percentage win 🥇